code/game/machinery/bank_machine.dm

```diff
@@ -0,0 +1,76 @@
+/obj/machinery/computer/bank_machine
+	name = "bank machine"
+	desc = "A machine used to deposit and withdraw station funds."
+	icon = 'goon/icons/obj/goon_terminals.dmi'
+	icon_state = "atm"
+	idle_power_usage = 100
+	var/siphoning = FALSE
+	var/last_warning = 0
+
+/obj/machinery/computer/bank_machine/attackby(obj/item/I, mob/user)
+	var/value = 0
+	if(istype(I, /obj/item/stack/spacecash))
+		var/obj/item/stack/spacecash/C = I
+		value = C.value
+	if(istype(I, /obj/item/weapon/coin))
+		var/obj/item/weapon/coin/C  = I
+		value = C.value
+	if(value)
+		SSshuttle.points += value
+		user << "<span class='notice'>You deposit [I]. The station now has [SSshuttle.points] credits.</span>"
+		qdel(I)
+		return
+	return ..()
+
+
+/obj/machinery/computer/bank_machine/update_icon()
+	if(stat & BROKEN)
+		icon_state = "atmb"
+	else if(stat & NOPOWER)
+		icon_state = "atmoff"
+	else
+		icon_state = "atm"
+
+/obj/machinery/computer/bank_machine/process()
+	..()
+	if(siphoning)
+		if(SSshuttle.points < 200)
+			say("Station funds depleted. Halting siphon.")
+			siphoning = FALSE
+		else
+			new /obj/item/stack/spacecash/c200(get_turf(src))
+			playsound(src.loc, 'sound/items/poster_being_created.ogg', 100, 1)
+			SSshuttle.points -= 200
+			if(last_warning < world.time && prob(15))
+				var/area/A = get_area(loc)
+				minor_announce("Unauthorized credit withdrawal underway in [A.map_name]." , "Network Breach")
```

Comment:
This is an "alert" type notice (something bad happened), so you should add 1 or TRUE as the third argument here so it uses the warning sound instead of the ding sound.